MCprep_addon/conf.py

```diff
@@ -137,30 +134,38 @@ def __init__(self):
 		# that no reading has occurred. If lib not found, will update to [].
 		# If ever changing the resource pack, should also reset to None.
 		self.material_sync_cache: List = []
-	
+
 		# Whether we use PO files directly or use the converted form
 		self.use_direct_i18n = False
 		# i18n using Python's gettext module
 		#
 		# This only runs if translations.py does not exist
+		self.languages: dict[str, gettext.NullTranslations] = {}
+		self._load_translations()
+
+	def _load_translations(self, prefix="") -> None:
+		"""Loads in mo file translation maps"""
 		try:
 			if not self.translations.exists():
-				self.languages: dict[str, gettext.NullTranslations] = {}
+				print(prefix, "No translations.py, running in location: ", __file__, " with ", self.languages_folder)
 				for language in self.languages_folder.iterdir():
-					self.languages[language.name] = gettext.translation("mcprep", 
-											 self.languages_folder, 
-											 fallback=True,
-											 languages=[language.name])
-			self.use_direct_i18n = True
-			self.log("Loaded direct i18n!")
-
-		except Exception:
+					print(prefix, "In langdir:", language.name)
+					self.languages[language.name] = gettext.translation(
+						"mcprep",
+						localedir=self.languages_folder,
+						fallback=True,
+						languages=[language.name])
```

Comment:
Hey @StandingPadAnimations are we sure this function is working well in Blender? I'm stuck here, can't get it to work properly in the test suite, despite confirming I'm pointing it to the right folder location and everything. Translations work in my blender instance, but that's because the translations.py file shows up (but if we use that, then we can't do direct tests for translations working below, which is the point).

This is the internal error that it runs into when calling the `self.languages[language.name] = gettext.translation(` line:

```
======================================================================
ERROR: test_translations (addon_test.AddonTest.test_translations)
Safely ensures translations are working fine still.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/patrickcrawford/Documents/gits/mcprep/test_files/addon_test.py", line 53, in test_translations
    self._test_translation()
  File "/Users/patrickcrawford/Documents/gits/mcprep/test_files/addon_test.py", line 75, in _test_translation
    test_env._load_translations(prefix=">>>\t")
  File "/Users/patrickcrawford/Documents/gits/mcprep/MCprep_addon/conf.py", line 165, in _load_translations
    raise e
  File "/Users/patrickcrawford/Documents/gits/mcprep/MCprep_addon/conf.py", line 153, in _load_translations
    self.languages[language.name] = gettext.translation(
                                    ^^^^^^^^^^^^^^^^^^^^
  File "/Applications/blender 4.2/Blender.app/Contents/Resources/4.2/python/lib/python3.11/gettext.py", line 538, in translation
    t = _translations.setdefault(key, class_(fp))
                                      ^^^^^^^^^^
  File "/Applications/blender 4.2/Blender.app/Contents/Resources/4.2/python/lib/python3.11/gettext.py", line 272, in __init__
    self._parse(fp)
  File "/Applications/blender 4.2/Blender.app/Contents/Resources/4.2/python/lib/python3.11/gettext.py", line 424, in _parse
    catalog[str(msg, charset)] = str(tmsg, charset)
                                 ^^^^^^^^^^^^^^^^^^
LookupError: unknown encoding: CHARSET
```

Reply:
I think I've traced the source of the problem
https://github.com/Moo-Ack-Productions/MCprep/blob/b72a7a38bb79ae70ebffb464751344182ac4cfe6/MCprep_addon/MCprep_resources/Languages/en_US/LC_MESSAGES/mcprep.po#L17

Try replacing `CHARSET` with `UTF-8` (to match the Chinese and Russian translations)

Reply:
Legendary! Thanks a mill, I'll clean this branch up and formally set to review in a bit - need to step away

```
-------------------------------------------------------------------------------
bversion   	ran_tests	ran	skips	failed	errors
-------------------------------------------------------------------------------
(4.2.0)   	test_translations	1	0	0	No errors

```